Signed-off-by: Kishore  Darmaradje <Kishore_Darmaradje@comcast.com>
---
diff --git a/FrontPanel/FrontPanel.cpp b/FrontPanel/FrontPanel.cpp
index e8a1dc5a..25bd6bf3 100644
--- a/FrontPanel/FrontPanel.cpp
+++ b/FrontPanel/FrontPanel.cpp
@@ -67,6 +67,7 @@
 #define API_VERSION_NUMBER_MINOR 0
 #define API_VERSION_NUMBER_PATCH 6
 
+extern "C" void __gcov_exit();
 namespace
 {
 
@@ -224,6 +225,7 @@ namespace WPEFramework
             patternUpdateTimer.Revoke(m_updateTimer);
 
             DeinitializeIARM();
+            __gcov_exit();
         }
         void FrontPanel::powerModeChange(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
         {
