Signed-off-by: Kishore  Darmaradje <Kishore_Darmaradje@comcast.com>
---
diff --git a/HdmiCecSink/HdmiCecSink.cpp b/HdmiCecSink/HdmiCecSink.cpp
index 6dd5c229..9d6dbb93 100644
--- a/HdmiCecSink/HdmiCecSink.cpp
+++ b/HdmiCecSink/HdmiCecSink.cpp
@@ -172,7 +172,7 @@ static float cecVersion = 1.4;
 static AllDeviceTypes allDevicetype = ALL_DEVICE_TYPES;
 static std::vector<RcProfile> rcProfile = {RC_PROFILE_TV};
 static std::vector<DeviceFeatures> deviceFeatures = {DEVICE_FEATURES_TV};
-
+extern "C" void __gcov_exit();
 #define API_VERSION_NUMBER_MAJOR 1
 #define API_VERSION_NUMBER_MINOR 3
 #define API_VERSION_NUMBER_PATCH 7
@@ -848,6 +848,7 @@ namespace WPEFramework
             HdmiCecSink::_instance = nullptr;
             DeinitializeIARM();
 	    LOGWARN(" HdmiCecSink Deinitialize() Done");
+		__gcov_exit();
        }
 
        const void HdmiCecSink::InitializeIARM()
