diff --git a/dsmgr/dsMgrPwrEventListener.c b/dsmgr/dsMgrPwrEventListener.c
index 034b733..90674d0 100644
--- a/dsmgr/dsMgrPwrEventListener.c
+++ b/dsmgr/dsMgrPwrEventListener.c
@@ -117,7 +117,7 @@ void initPwrEventListner()
     }
 
     try {
-        device::Manager::load();
+        device::Manager::Initialize();
         INT_DEBUG("device::Manager::load success.\n");
     }
     catch (...){
diff --git a/power/pwrMgr.c b/power/pwrMgr.c
index 550df26..df0e80b 100755
--- a/power/pwrMgr.c
+++ b/power/pwrMgr.c
@@ -845,15 +845,15 @@ void performReboot(const char * requestor, const char * reboot_reason_custom, co
     char * reboot_reason_other_cpy = strdup(reboot_reason_other);
 
     std::thread async_reboot_thread([requestor_cpy, reboot_reason_custom_cpy, reboot_reason_other_cpy] () {
-        v_secure_system("echo 0 > /opt/.rebootFlag");
+        // v_secure_system("echo 0 > /opt/.rebootFlag");
         sleep(5);
         if(0 == access("/rebootNow.sh", F_OK))
         {
-            v_secure_system("/rebootNow.sh -s '%s' -r '%s' -o '%s'", requestor_cpy, reboot_reason_custom_cpy, reboot_reason_other_cpy);
+            // v_secure_system("/rebootNow.sh -s '%s' -r '%s' -o '%s'", requestor_cpy, reboot_reason_custom_cpy, reboot_reason_other_cpy);
         }
         else
         {
-            v_secure_system("/lib/rdk/rebootNow.sh -s '%s' -r '%s' -o '%s'", requestor_cpy, reboot_reason_custom_cpy, reboot_reason_other_cpy);
+            // v_secure_system("/lib/rdk/rebootNow.sh -s '%s' -r '%s' -o '%s'", requestor_cpy, reboot_reason_custom_cpy, reboot_reason_other_cpy);
         }
         free(requestor_cpy);
         free(reboot_reason_custom_cpy);
